mycroft/enclosure/gui.py

```diff
@@ -0,0 +1,248 @@
+            else:
+                raise FileNotFoundError("Unable to find page: {}".format(name))
+
+        self.skill.bus.emit(Message("gui.page.show",
+                                    {"page": page_urls,
+                                     "index": index,
+                                     "__from": self.skill.skill_id,
+                                     "__idle": override_idle}))
+
+    def remove_page(self, page):
+        """ Remove a single page from the GUI.
+
+            Args:
+                page (str): Page to remove from the GUI
+        """
+        return self.remove_pages([page])
+
+    def remove_pages(self, page_names):
+        """ Remove a list of pages in the GUI.
+
+            Args:
+                page_names (list): List of page names (str) to display, such as
+                                   ["Weather.qml", "Forecast.qml", "Other.qml"]
+        """
+        if not isinstance(page_names, list):
+            raise ValueError('page_names must be a list')
+
+        # Convert pages to full reference
+        page_urls = []
+        for name in page_names:
+            page = self.skill.find_resource(name, 'ui')
+            if page:
+                page_urls.append("file://" + page)
+            else:
+                raise FileNotFoundError("Unable to find page: {}".format(name))
+
+        self.skill.bus.emit(Message("gui.page.delete",
+                                    {"page": page_urls,
+                                     "__from": self.skill.skill_id}))
+
+    def show_text(self, text, title=None, override_idle=None):
+        """ Display a GUI page for viewing simple text
+
+        Arguments:
+            text (str): Main text content.  It will auto-paginate
+            title (str): A title to display above the text content.
+        """
+        self.clear()
+        self["text"] = text
+        self["title"] = title
+        self.show_page("SYSTEM_TextFrame.qml", override_idle)
+
+    def show_image(self, url, caption=None,
+                   title=None, fill=None,
+                   override_idle=None):
+        """ Display a GUI page for viewing an image
+
+        Arguments:
+            url (str): Pointer to the image
+            caption (str): A caption to show under the image
+            title (str): A title to display above the image content
+            fill (str): Fill type supports 'PreserveAspectFit',
+            'PreserveAspectCrop', 'Stretch'
+        """
+        self.clear()
+        self["image"] = url
+        self["title"] = title
+        self["caption"] = caption
+        self["fill"] = fill
+        self.show_page("SYSTEM_ImageFrame.qml", override_idle)
+
+    def show_html(self, html, resource_url=None, override_idle=None):
+        """ Display an HTML page in the GUI
+
+        Arguments:
+            html (str): HTML text to display
+            resource_url (str): Pointer to HTML resources
+        """
+        self.clear()
+        self["html"] = html
```

Comment:
I have never seen the self['name'] notation before.  what does this do?  is it different than self.name?

Reply:
This class implements the `__setitem__()` magic method to access a member dictionary and send over the bus. using self['name'] will trigger the aforementioned method.

Reply:
Interesting.  Not sure I am a fan of this implementation, but that is a problem for another day.